inhouse_bot/tournament/tournament_check_job.py

```diff
@@ -1,15 +1,81 @@
 import logging
+import random
 from discord.ext import commands
-from typing import Optional
+from typing import List, Optional, Tuple
+from inhouse_bot.common_utils.lol_api.tasks import (
+    get_match_info_by_id,
+    get_tournament_match_history,
+)
 
 from inhouse_bot.database_orm import session_scope
+from inhouse_bot.database_orm.session.session import Session
+from inhouse_bot.database_orm.tables.game import Game
+from inhouse_bot.database_orm.tables.tournament import Tournament
 
+from inhouse_bot.tournament.tournament_handler import tournament_handler
 
-async def tournament_check(bot: commands.Bot, server_id: Optional[int]):
+
+def get_active_games(session: Session) -> List[Tuple[Game, Tournament]]:
+    return (
+        session.query(Game, Tournament)
+        .select_from(Game)
+        .outerjoin(Tournament)
+        .filter(Game.winner == None)
+        .all()
+    )
+
+
+async def tournament_check(bot: commands.Bot, server_id: int):
     """
-    Tournaments API background job. This looks at the active tournament matches
-    and check for status updates on them.
+    Tournaments background job. This looks at the active games and tries to find a completed tournament match
     """
 
     logging.info("Checking for completed matches")
-    # with session_scope() as session:
+    with session_scope() as session:
+
+        # query for tournaments whose games do not have a winner
+        for (game, tournament) in get_active_games(session):
+            logging.info(f"Game {game.id} is still active. Checking for updates.")
+
+            start_timestamp = int(game.start.timestamp())
+
+            # Find a player in the game and search their tournament match history after the game start time
+            # The player is chosen at random because it doesn't matter which player we use, but some players
+            # may have played more tournament matches than others.
+            player_puuid = random.choice(

```

Comment:
instead of random, why not just get the first player in the game? what value does random provide you here?

Reply:
Probably overthought this one. My reasoning was that if the first player somehow played many tournament games, the check logic would have to fetch many matches and might not find the particular game that hasn't been scored. But randomly choosing a player in the mix to search on makes those odds better while also not overloading the system by checking every player's match history since the game start time.